File: Tarumt.CC.Ecommerce.Admin.FE/src/app/types/responses/product-response.ts
import { ApiResponseBase } from "./api-response.response";
import { ProductCategoryResponse } from "./product-category-response";

export interface ProductResponse extends ApiResponseBase {
  name: string;
  short_name: string;
  count: number;
  price: number;
  discount_rate: number;
  description: string;
  image_url: string;
  categories: ProductCategoryResponse[];
  start_at: string;
  expired_at: string;
}

File: Tarumt.CC.Ecommerce.SharedLibarary/Infrastructure/Responses/ProductResponse.cs
using Tarumt.CC.Ecommerce.SharedLibarary.Infrastructure.Responses;

namespace Ky.Web.CMS.SharedLibarary.Infrastructure.Responses
{
    public class ProductResponse : Response
    {
        public required string Name { get; set; }

        public required string ShortName { get; set; }

        public required int Count { get; set; }

        public required decimal Price { get; set; }

        public required decimal DiscountRate { get; set; }

        public required string Description { get; set; }

        public required string ImageUrl { get; set; }

        public required ProductCategoryResponse[] Categories { get; set; }

        public DateTime? StartAt { get; set; }

        public DateTime? ExpiredAt { get; set; }
    }
}
